jobs/ads-incrementality-dap-collector/ads_incrementality_dap_collector/main.py

```diff
@@ -0,0 +1,59 @@
+import click
+import logging
+import traceback
+
+from constants import LOG_FILE_NAME
+from helpers import get_config, get_experiment, prepare_results_rows, collect_dap_results, write_job_logs_to_bucket, write_results_to_bq
+
+@click.command()
+@click.option("--gcp_project", help="GCP project id", required=True)
+@click.option("--job_config_bucket", help="GCS bucket where the configuration for this job can be found. See example_config.json for format details.", required=True)
+@click.option(
+    "--hpke_token",
+    envvar='DAP_HPKE_TOKEN',
+    help="The token defined in the collector credentials, used to authenticate to the leader",
+    required=True,
+)
+@click.option(
+    "--hpke_private_key",
+    envvar='DAP_PRIVATE_KEY',
+    help="The private key defined in the collector credentials, used to decrypt shares from the leader and helper",
+    required=True,
+)
+@click.option(
+    "--batch_start",
+    type=int,
+    envvar='BATCH_START',
+    help="Start of the collection interval, as the number of seconds since the Unix epoch",
+    required=True,
+)
+def main(gcp_project, job_config_bucket, hpke_token, hpke_private_key, batch_start):
+    try:
+        logging.info(f"Starting collector job with configuration from gcs bucket: {job_config_bucket}")
+        config = get_config(gcp_project, job_config_bucket, hpke_token, hpke_private_key, batch_start)
+        logging.info(f"Starting collector job for experiments: {config.nimbus.experiments}.")
+
+        for experiment_config in config.nimbus.experiments:
+            experiment = get_experiment(experiment_config.slug, config.nimbus.api_url)
+
+            tasks_to_collect = prepare_results_rows(experiment)
+
+            collected_tasks = collect_dap_results(tasks_to_collect, config.dap, experiment_config)
+
+            write_results_to_bq(collected_tasks, config.bq)
+
+            write_job_logs_to_bucket(gcp_project, job_config_bucket)
+    except Exception as e:
```

Comment:
With the way this exception handling is structured, if we fail to collect on a particular experiment, we'll break out of the loop that process experiments and end the script. 

I think it could be good to change the structure so that if we fail on one experiment, the job could continue on and try processing the other experiments ...

(Leaving this is a note to myself while I work through the other comments)

Reply:
Done.